keyboards/basekeys/slice/rev1_rgb/keymaps/via/keymap.c

```diff
@@ -0,0 +1,163 @@
+  //|------------------+--------+--------+--------+--------+--------+--------|   |--------+--------+--------+--------+--------+--------+----------+----------|
+                RGB_TOG, KC_LCTL, KC_LALT, KC_LALT,  KC_SPC,  KC_SPC,  KC_SPC,      KC_SPC,  KC_SPC,                            KC_RALT,   KC_RCTL,   MO(_FN)
+  //`------------------------------------------------------------------------|   |---------------------------------------------------------------------------'
+  ),
+
+  [_RAISE] = LAYOUT_all(
+  //,------------------------------------------------------------------------|   |---------------------------------------------------------------------------.
+       KC_ESC, KC_GRAVE,    KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,        KC_7,    KC_8,    KC_9,    KC_0, KC_MINS,  KC_EQL,      KC_BSPACE,
+  //|------------------+--------+--------+--------+--------+--------+--------|   |--------+--------+--------+--------+--------+--------+----------+----------|
+       RGB_MODE_FORWARD,  KC_TAB,    KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,        KC_Y,    KC_U,    KC_I,    KC_O,    KC_P, KC_LBRC,   KC_RBRC,   KC_BSLS,
+  //|------------------+--------+--------+--------+--------+--------+--------|   |--------+--------+--------+--------+--------+--------+----------+----------|
+      RGB_MODE_GRADIENT, KC_CAPS,    KC_A,    KC_S,    KC_D,    KC_F,    KC_G,        KC_H,    KC_J,    KC_K,    KC_L, KC_SCLN, KC_QUOT,        KC_ENT,
+  //|------------------+--------+--------+--------+--------+--------+--------|   |--------+--------+--------+--------+--------+--------+----------+----------|
+          RGB_MODE_XMAS, KC_LSFT,    KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,        KC_B,    KC_N,    KC_M, KC_COMM,  KC_DOT, KC_SLSH,   KC_RSFT,   KC_RSFT,
+  //|------------------+--------+--------+--------+--------+--------+--------|   |--------+--------+--------+--------+--------+--------+----------+----------|
+                RGB_TOG, KC_LCTL, KC_LALT, KC_LALT,  KC_SPC,  KC_SPC,  KC_SPC,       KC_SPC,  KC_SPC,                            KC_RALT,   KC_RCTL,   MO(_FN)
+  //`------------------------------------------------------------------------|   |---------------------------------------------------------------------------'
+  )
+};
+
+
+int RGB_current_mode;
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  bool result = false;
+  switch (keycode) {
+    #ifdef RGBLIGHT_ENABLE
+      case RGB_MOD:
+          if (record->event.pressed) {
+            rgblight_mode(RGB_current_mode);
+            rgblight_step();
+            RGB_current_mode = rgblight_get_mode();
+          }
+        break;
+      case RGB_RST:
+          if (record->event.pressed) {
+            eeconfig_update_rgblight_default();
+            rgblight_enable();
+            RGB_current_mode = rgblight_get_mode();
+          }
+        break;
+    #endif
+    default:
+      result = true;
+      break;
+  }
+
+  return result;
+}
```

Comment:
However, if you want to do more advanced stuff, this may be better.
```suggestion
bool process_record_user(uint16_t keycode, keyrecord_t *record) {
  switch (keycode) {
#ifdef RGBLIGHT_ENABLE
      case RGB_RST:
          if (record->event.pressed) {
            eeconfig_update_rgblight_default();
            rgblight_enable();
            RGB_current_mode = 0;
          }
        break;
#endif
  }
  return true;
}

void post_process_record_user(uint16_t keycode, keyrecord_t *record) {
    if (!record->event.pressed && (keycode == RGB_MOD || keycode == RGB_RMOD)) {
        RGB_current_mode = rgblight_get_mode();
    }
}
```

But honestly, I think all you need is: 
```suggestion
bool process_record_user(uint16_t keycode, keyrecord_t *record) {
    switch (keycode) {
#ifdef RGBLIGHT_ENABLE
        case RGB_RST:
            if (record->event.pressed) {
              eeconfig_update_rgblight_default();
              rgblight_enable();
            }
          break;
#endif
    }
    return true;
}
```

Reply:
All of this was removed.